src/modules/iotjs_module_adc.c

```diff
@@ -257,15 +263,22 @@ JHANDLER_FUNCTION(Close) {
 
   if (jcallback) {
     ADC_ASYNC(close, adc, jcallback, kAdcOpClose);
-  } else {
-    if (!iotjs_adc_close(adc)) {
-      JHANDLER_THROW(COMMON, "ADC Close Error");
-    }
+  } else if (!iotjs_adc_close(adc)) {
```

Comment:
`adc.close` is async function. It should run asynchronously, even if there is no `jcallback`.